utils/build_profiler_report.py

```diff
@@ -0,0 +1,166 @@
+        return '%.1fs' % t
+    if t < 60 * 60:
+        return '%dm%.1fs' % (t / 60, t % 60)
+    return '%dh%dm%.1fs' % (t / (60 * 60), t % (60 * 60) / 60, t % 60)
+
+
+def GenerateWebTreeMap(CurrentDict: dict, BaselineDict: dict) -> None:
+    """Create file for webtreemap to generate an HTML from; if target is new, append _NEW"""
+    with open(sys.argv[1] + ".webtreemap", 'w') as fp:
+        for target in CurrentDict.keys():
+            new_tag = ''
+            if BaselineDict and target not in BaselineDict.keys():
+                new_tag = '_NEW'
+
+            fp.write(str(CurrentDict[target]) + ' ' + target
+                     + '_' + FormatTime(CurrentDict[target]) + new_tag + '\n')
+
+
+def GetTotalTime(TargetDurationDict: dict) -> int:
+    """Return sum of durations for all targets in dict"""
+    total_time = 0
+    for target in TargetDurationDict.keys():
+        total_time += TargetDurationDict[target]
+    return total_time
+
+
+def GetTotalTimeIntersect(TargetDurationDict_A: dict, TargetDurationDict_B: dict) -> int:
+    """Return sum of durations for targets in A that are also in B"""
+    total_time = 0
+    for target in TargetDurationDict_A.keys():
+        if target in TargetDurationDict_B.keys():
+            total_time += TargetDurationDict_A[target]
+    return total_time
+
+
+def PrintReport(CurrentDict: dict, BaselineDict: dict = None) -> None:
+    """Print report with results of profiling to stdout"""
+
+    # If the targets have changed between baseline and current, we are using total_time_intersect
+    # to calculate delta (ratio of durations of targets) instead of total_time
+    if BaselineDict and BaselineDict.keys() != CurrentDict.keys():
+        print("Warning: the targets in the current logfile differ from those in the baseline\
+             logfile; therefore the Δ delta is calculated without taking the added/removed\
+                  targets into consideration. If the added/removed targets modify the behavior of\
+                      targets in both logfiles, the Δ delta may not make sense.\n-----")
+        target_mismatch = True
+        total_time_current_intersect = GetTotalTimeIntersect(CurrentDict, BaselineDict)
+        total_time_baseline_intesect = GetTotalTimeIntersect(BaselineDict, CurrentDict)
+    else:
+        target_mismatch = False
+
+    header = [f'{"Target:":60}', f"{'%':4}", f"{'Δ':5}", f"{'T':8}", f"{'%Δ':5}", "Note"]
+    print(' | '.join(header))
+
+    total_time_current = GetTotalTime(CurrentDict)
+    if BaselineDict:
+        total_time_baseline = GetTotalTime(BaselineDict)
+
+    # sort targets by % taken of build time
+    temp = {k: v for k, v in sorted(CurrentDict.items(), key=lambda item: item[1], reverse=True)}
+    CurrentDict = temp
+
+    for target in CurrentDict.keys():
+        # percentage of build time that the target took
+        perc = CurrentDict[target]/total_time_current * 100
+
+        # difference between perc in current and in baseline
```

Comment:
Maybe a difference in seconds would be more interesting or interesting as well?

Reply:
good idea, the script now prints the difference in seconds as well in addition to the percentage change in time